### PR TITLE
Use only first line of commit when creating referenced comment

### PR DESCRIPTION
**Pull request by CirnoT**
We should only include first line. Ideally we'd use some form of ellipsis but that would require some heavy refactoring, as right now we store anchor HTML in database.

Before:
chrome_2020-06-18_14-00-34

After:
chrome_2020-06-18_14-00-45

### CODE REVIEW

**Comment by CirnoT** [comment]
The alternative here is to write a migration that extracts actual content from anchor link in database and save it without it, then we render it as we do other commits with ellipsis

**Review by lafriks — APPROVED** [review]

**Review by 6543 — APPROVED** [review]
would say this is a bugfix witch we can backport to v1.12.1

**Comment by 6543** [comment]
@CirnoT beside this you could refactor it for v1.13 in a new pull :)

**Comment by CirnoT** [comment]
@6543 If we do backport, we backport is instead of this PR, otherwise we'll lose additional data for new commits.

**Comment by CirnoT** [comment]
Alternatively, we could merge this for v1.12.1 backport and refactor to store only commit hash in database and fetch commit message on-the-fly for v1.13.

**Comment by silverwind** [comment]
Can you also fix this on the frontpage new feed?

image

**Comment by 6543** [comment]
I would display the first line and make a a "> more" in the next line
witch works similar to the "Downloads" on the release page?

**Comment by silverwind** [comment]
> make a a "> more"

We do support a `...` button on the repo summary page for the top commit. That said I think it's fine to not have such button everywhere where a commit can appear.

**Comment by mrsdizzie** [comment]
Think it would look bad and be clunky to have any kind of expanding 'read more' button. Thats why it already links to the commit page : )

File: modules/repofiles/action.go [hunk 1]
```diff
@@ -111,7 +112,7 @@ func UpdateIssuesCommit(doer *models.User, repo *models.Repository, commits []*r
 				continue
 			}
 
-			message := fmt.Sprintf(`<a href="%s/commit/%s">%s</a>`, repo.Link(), c.Sha1, html.EscapeString(c.Message))
+			message := fmt.Sprintf(`<a href="%s/commit/%s">%s</a>`, repo.Link(), c.Sha1, html.EscapeString(strings.Split(c.Message, "\n")[0]))
```
Comment (zeripath) [inline modules/repofiles/action.go]:
```suggestion
			message := fmt.Sprintf(`<a href="%s/commit/%s">%s</a>`, repo.Link(), c.Sha1, html.EscapeString(strings.SplitN(c.Message, "\n", 2)[0]))
```

**Comment by 6543** [comment]
whats the point for SplitN since the rest will be cut away in all cases? - does it use les memory?

**Comment by 6543** [comment]
@zeripath whats the point for SplitN since the rest will be cut away in all cases? - does it use less memory?

**Comment by 6543** [comment]
> Think it would look bad and be clunky to have any kind of expanding 'read more' button. Thats why it already links to the commit page : )

good point - so i think we dont need any refactor just display only the first line

**Comment by CirnoT** [comment]
>Can you also fix this on the frontpage news feed maybe?

Definitely should, let me take a quick look

**Comment by zeripath** [comment]
@6543 what's the point of reading the rest of the string splitting it in to every line when you don't care about the other lines?

SplitN reads until it has the requested number of splits and stops. Split reads the entire string.

**Comment by CirnoT** [comment]
Should be fixed on feeds now too

**Review by silverwind — APPROVED** [review]
Frontpage working, thanks!

**Pull request merged**